dcos/dcos-ui
# DCOS-50724: eliminate CompositeState dep from NodeBreadcrumbs

Eliminate dependency on CompositeState by implementing a ~custom hook~ higher order component to pull latest summary from MesosSummaryStore.

This implementation makes use of the mesos state summary we are already polling for (globally) every 2s so that we don't have to make any extra network requests.

Some random notes I used to try to better understand the constraints [here](https://gist.github.com/natmegs/16f9fbd51a9ce6835f9c43a663ba875a).

## Testing

- Go to Nodes page
- Click on a node to go to its detail page
- Ensure breadcrumbs at top of page show Nodes > <node hostname>

## Trade-offs

- The JIRA specifies to use RxJS but this implementation does not. I found that an RxJS solution wasn't super intuitive for this particular case because the MesosSummaryStore uses an event emitter pattern rather than a stream-based request pattern. We could wrap the emitters to emit to a `Subject` and compose a stream from that (which is what I tried initially), but it involved a lot of overhead.
- When we are on the nodes page, not only do we poll for the mesos summary, but we also poll 2 different nodes endpoints every 2s (`/system/health/v1/nodes` and `/net/v1/nodes`). It seems unfortunate that we can't make one request to get the nodes information that we need 

## Dependencies

N/A

## Screenshots

N/A
